### PR TITLE
Fix #8830: Added the policyProvided option for the CSP context-param

### LINKED ISSUE #8830: CSP: CSP support for Portlet implementation
### Description

We've been using Primefaces in our Liferay application for a while now. From security standpoints we would like to tighten our CSP Header, removing the 'unsafe-inline' tag.

There already is a primefaces.CSP context-param that enables CSP for Primefaces. This will rewrite the inline Javascript and set the CSP HTTP header. Since we only use Primefaces in a Portlet context the excisting primefaces.CSP context param was not sufficient. We would like Primefaces to rewrite the inline event-handlers, but still provide the CSP header and nonce ourself.

### Describe the solution you would like

We would like to add a context-param to declare that we implement the CSP Header ourselfs. Using this param Primefaces should only do the Eventhandler rewriting and skip setting the CSP Header and Nonce.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by MijnRVO-general**
We've been using Primefaces in our Liferay application for a while now. From security standpoints we would like to tighten our CSP Header, removing the 'unsafe-inline' tag. 

There already is a primefaces.CSP context-param that enables CSP for Primefaces. This will rewrite the inline Javascript and set the CSP HTTP header. Since we only use Primefaces in a Portlet context the excisting primefaces.CSP context param was not sufficient. We would like Primefaces to rewrite the inline event-handlers, but still provide the CSP header and nonce ourself. 

In this pull request we've added the `policyProvided` option for the CSP context-param. Setting this param will skip initialising the CSP Header and Nonce while still rewriting the inline JavaScript.

### CODE REVIEW

**Comment by melloware** [comment]
Can you open an issue first so we can link this PR to it.

File: primefaces/src/main/java/org/primefaces/config/PrimeConfiguration.java [hunk 1]
```diff
@@ -55,6 +55,7 @@ public class PrimeConfiguration {
     private final boolean earlyPostParamEvaluation;
     private final boolean moveScriptsToBottom;
     private boolean csp;
+    private boolean policyProvided;
     private String cspPolicy;
```
Comment (melloware) [inline primefaces/src/main/java/org/primefaces/config/PrimeConfiguration.java:59]: Please update `docs\12_0_0\gettingstarted\configuration.md` and `\docs\12_0_0\core\contentsecuritypolicy.md` to update with all this new info

**Review by melloware — CHANGES_REQUESTED** [review]

**Comment by melloware** [comment]
Also it looks like the build is failing unit tests.

**Comment by MijnRVO-general** [comment]
I've opened an issue: https://github.com/primefaces/primefaces/issues/8830

File: primefaces/src/main/java/org/primefaces/config/PrimeConfiguration.java (hunk 1 above)
Comment (MijnRVO-general) [reply primefaces/src/main/java/org/primefaces/config/PrimeConfiguration.java:59]: I've added the documentation as requested

File: docs/12_0_0/core/contentsecuritypolicy.md [hunk 2]
```diff
@@ -34,6 +34,20 @@ Browsers will report CSP violations to a user-defined HTTP endpoint using `repor
...
+
+```xml
+<context-param>
+    <param-name>primefaces.CSP</param-name>
+    <param-value>policyProvided</param-value>
+</context-param>
+<context-param>
+    <param-name>primefaces.CSP_REPORT_ONLY_POLICY</param-name>
+    <param-value>report-uri /my-csp-reports</param-value>
+</context-param>
```
Comment (melloware) [inline docs/12_0_0/core/contentsecuritypolicy.md]: Can you remove this CSP_REPORT_ONLY_POLICY here in your doc as you don't need it right for your `policyProvided` type right?

**Review by melloware — CHANGES_REQUESTED** [review]

File: docs/12_0_0/core/contentsecuritypolicy.md [hunk 3]
```diff
@@ -34,6 +34,20 @@ Browsers will report CSP violations to a user-defined HTTP endpoint using `repor
...
+Primefaces. Using `policyProvided` will make the code CSP complient while not setting the CSP Header.
+
+```xml
+<context-param>
+    <param-name>primefaces.CSP</param-name>
+    <param-value>policyProvided</param-value>
+</context-param>
+<context-param>
+    <param-name>primefaces.CSP_REPORT_ONLY_POLICY</param-name>
+    <param-value>report-uri /my-csp-reports</param-value>
```
Comment (melloware) [inline docs/12_0_0/core/contentsecuritypolicy.md]: Actually if you do need REPORT_ONLY_POLICY shouldn't we make this a real world example how you would use it with Portlets?

File: docs/12_0_0/core/contentsecuritypolicy.md (hunk 2 above)
Comment (MijnRVO-general) [reply docs/12_0_0/core/contentsecuritypolicy.md]: Yes, this was a copy-paste mistake, sorry.

File: docs/12_0_0/core/contentsecuritypolicy.md (hunk 3 above)
Comment (MijnRVO-general) [reply docs/12_0_0/core/contentsecuritypolicy.md]: The REPORT_ONLY_POLICY is not required, it was a copy-paste error.

**Comment by melloware** [comment]
So how do you set the NONCE in your Portlet environement?  It would be great if you updated the doc to explain HOW to use this in the Portlet environment for others to use.

**Comment by MijnRVO-general** [comment]
We use Liferay as our CMS, and have a custom-made CSP-solution (provided by Liferay), as Liferay 7.1 does not support CSP yet. It might arrive in 7.4. So the exact situation is not the best example perhaps. We've kept it in general terms but this hopefully clarifies it.

File: docs/12_0_0/core/contentsecuritypolicy.md [hunk 4]
```diff
@@ -34,6 +34,18 @@ Browsers will report CSP violations to a user-defined HTTP endpoint using `repor
 </context-param>
 ```
 
+When using Primefaces in a component (like portlets for intance), the platform hosting the component might need to
```
Comment (jepsar) [inline docs/12_0_0/core/contentsecuritypolicy.md]:
```suggestion
When using PrimeFaces in a component (like portlets for instance), the platform hosting the component might need to
```

File: docs/12_0_0/core/contentsecuritypolicy.md [hunk 5]
```diff
@@ -34,6 +34,18 @@ Browsers will report CSP violations to a user-defined HTTP endpoint using `repor
...
+manage the CSP headers and nonce on scripts. For instance in a CMS where different kinds of components can be used
+and all need to have the same nonce. In these cases the option `policyProvided` can be used. This will make
+Primefaces CSP compliant, while not setting the CSP Header and add the nonce to javascripts.
```
Comment (jepsar) [inline docs/12_0_0/core/contentsecuritypolicy.md]:
```suggestion
PrimeFaces CSP compliant, while not setting the CSP Header and add the nonce to JavaScripts.
```

**Review by melloware — APPROVED** [review]

**Comment by tandraschko** [comment]
will review early next week, i'm on weekend now :D

**Pull request merged**